google/cloud/spanner/internal/compiler_info_test.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/internal/compiler_info.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+namespace {
+
+TEST(BuildInfo, CompilerId) {
+  auto cn = CompilerId();
+  EXPECT_FALSE(cn.empty());
+  EXPECT_THAT(cn, ::testing::Not(::testing::HasSubstr("@")));
```

Comment:
All these tests ensuring that "@" does not appear in the string were to ensure that none of the `@FOO@` cmake variables slipped through without being substituted. Since this is now only testing compiler_info, which doesn't use any cmake substitution variables, we can probably get rid of all of these expectations.

Reply:
Thanks.